be/src/exec/vectorized/sorting/merge_column.cpp

```diff
@@ -0,0 +1,354 @@
+
+                for (int col = 0; col < sort_desc.num_columns(); col++) {
+                    const Column* left_col = left_run.get_column(col);
+                    const Column* right_col = right_run.get_column(col);
+                    MergeTwoColumn merge2(sort_desc.get_column_desc(col), left_col, right_col, &equal_ranges, output);
+                    Status st = left_col->accept(&merge2);
+                    CHECK(st.ok());
+                    if (equal_ranges.size() == 0) {
+                        break;
+                    }
+                }
+            }
+        }
+
+        return Status::OK();
+    }
+
+private:
+    // Check if two run has intersect, if not we don't need to merge them row by row
+    // @return 0 if two run has intersect, -1 if left is less than right, 1 if left is greater than right
+    static int run_intersect(const SortDescs& sort_desc, const SortedRun& left_run, const SortedRun& right_run) {
+        // Compare left tail with right head
+        int left_tail_cmp = compare_chunk_row(sort_desc, left_run.orderby, right_run.orderby, left_run.range.second - 1,
+                                              right_run.range.first);
+        if (left_tail_cmp < 0) {
+            return -1;
+        }
+
+        // Compare left head with right tail
+        int left_head_cmp = compare_chunk_row(sort_desc, left_run.orderby, right_run.orderby, left_run.range.first,
+                                              right_run.range.second - 1);
+        if (left_head_cmp > 0) {
+            return 1;
+        }
+        return 0;
+    }
+};
+
+Status merge_sorted_chunks_two_way(const SortDescs& sort_desc, const ChunkPtr left, const ChunkPtr right,
+                                   Permutation* output) {
+    DCHECK_LE(sort_desc.num_columns(), left->num_columns());
+    DCHECK_LE(sort_desc.num_columns(), right->num_columns());
+
+    SortedRun left_run(left, 0, left->num_rows());
+    SortedRun right_run(right, 0, right->num_rows());
+    return MergeTwoChunk::merge_sorted_chunks_two_way(sort_desc, left_run, right_run, output);
+}
+
+Status merge_sorted_chunks_two_way(const SortDescs& sort_desc, const SortedRun& left, const SortedRun& right,
+                                   Permutation* output) {
+    return MergeTwoChunk::merge_sorted_chunks_two_way(sort_desc, left, right, output);
+}
+
+Status merge_sorted_chunks_two_way(const SortDescs& sort_desc, const SortedRuns& left, const SortedRuns& right,
+                                   SortedRuns* output) {
+    CHECK(false) << "TODO";
+    return Status::NotSupported("TODO");
+}
+
+// Merge multiple chunks in two-way merge
+Status merge_sorted_chunks(const SortDescs& descs, const std::vector<ChunkPtr>& chunks, ChunkPtr* output) {
+    std::vector<ChunkPtr> current(chunks);
+    while (current.size() > 1) {
+        std::vector<ChunkPtr> next_level;
+        int level_size = current.size() & ~1;
```

Comment:
Will writing this improve performance?

Reply:
not for performance, this expression is equivalent to `current.size() - current.size() % 2`, but more compact